app/src/main/java/io/appium/uiautomator2/utils/Device.java

```diff
@@ -36,11 +36,30 @@ public static void wake() throws RemoteException {
         getUiDevice().wakeUp();
     }
 
-    public static void scrollTo(String scrollToString) throws UiObjectNotFoundException {
-        // TODO This logic needs to be changed according to the request body from the Driver
+    public static void scrollToElement(UiSelector selector, int maxSwipes)
+            throws UiObjectNotFoundException {
         UiScrollable uiScrollable = new UiScrollable(new UiSelector().scrollable(true).instance(0));
-        uiScrollable.scrollIntoView(new UiSelector().descriptionContains(scrollToString).instance(0));
-        uiScrollable.scrollIntoView(new UiSelector().textContains(scrollToString).instance(0));
+        String uiScrollableClassName = uiScrollable.getClassName();
+        String hScrollViewClassName = android.widget.HorizontalScrollView.class.getName();
+        int defaultMaxSwipes = uiScrollable.getMaxSearchSwipes();
+
+        if (java.util.Objects.equals(uiScrollableClassName, hScrollViewClassName)) {
+            uiScrollable.setAsHorizontalList();
+        }
+
+        if (maxSwipes > 0) {
+            uiScrollable.setMaxSearchSwipes(maxSwipes);
+        }
+
+        boolean scrollResult = uiScrollable.scrollIntoView(selector);
```

Comment:
can this throw an exception? Should we put the next operator into `finally` block?

Reply:
Yes, this actually might be a good idea. 
The declaration of **UiScrollable.scrollIntoView(...)** says that it can throw a **UiObjectNotFoundException**. I'll update the logic of **Device.scrollToElement()** accordingly.